File: leetcode/editor/en/[287]Find the Duplicate Number.java
287
        Find the Duplicate Number
        2022-12-30 13:02:28

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int findDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            if (set.contains(num)) {
                return num;
            }
            set.add(num);
        }
        return -1;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[287]Find the Duplicate Number.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function (nums) {
    const isDuplicated = new Array(nums.length).fill(false);

    for (const num of nums) {
        if (isDuplicated[num]) {
            return num;
        }

        isDuplicated[num] = true;
    }

    return -1;
};
//leetcode submit region end(Prohibit modification and deletion)
